### PR TITLE
fix sharding plugin configs that contain trigger plugin configuration

### PR DESCRIPTION
**Pull request by osherdp**
When opening openshift/release/pull/26022, I've noticed it lacking the ability to preserve trigger plugin configuration on the ``correctly-sharded-config`` job. This change should consider trigger configuration

It's dependent on this change in test-infra: 
kubernetes/test-infra/pull/25173

### CODE REVIEW

**Comment by osherdp** [comment]
/hold
waiting until test-infra's PR gets in

File: cmd/repo-init/main.go [hunk 1]
```diff
@@ -581,6 +581,10 @@ Ensure that webhooks are set up for Prow to watch GitHub state.`)
 		Repos:            []string{orgRepo},
 		ReviewActsAsLgtm: true,
 	})
+	pluginConfig.Triggers = append(pluginConfig.Triggers, plugins.Trigger{
+		Repos:       []string{orgRepo},
+		TrustedApps: []string{},
+	})
```
Comment (petr-muller) [inline cmd/repo-init/main.go:587]: I don't see why this would be necessary - why generate what is basically an empty config?

**Review by petr-muller — APPROVED** [review]

**Comment by petr-muller** [comment]
This will need a followup in o/release to avoid breaking PRs - ideally submit a PR in advance so that the period where PRs are broken in openshift/release is as short as possible.

```diff
 diff --git a/core-services/prow/02_config/_plugins.yaml b/core-services/prow/02_config/_plugins.yaml
index b7a15185..69d19175 100644
--- a/core-services/prow/02_config/_plugins.yaml
+++ b/core-services/prow/02_config/_plugins.yaml
@@ -1807,8 +1807,3 @@ size:
   xl: 0
   xxl: 0
 slack: {}
-triggers:
-- repos:
-  - openshift/assisted-service
-  trusted_apps:
-  - dependabot
diff --git a/core-services/prow/02_config/openshift/assisted-service/_pluginconfig.yaml b/core-services/prow/02_config/openshift/assisted-service/_pluginconfig.yaml
index 2fbd7a9f..4c8a58a3 100644
--- a/core-services/prow/02_config/openshift/assisted-service/_pluginconfig.yaml
+++ b/core-services/prow/02_config/openshift/assisted-service/_pluginconfig.yaml
@@ -38,3 +38,8 @@ plugins:
     plugins:
     - approve
     - size
+triggers:
+- repos:
+  - openshift/assisted-service
+  trusted_apps:
+  - dependabot 
```

**Comment by osherdp** [comment]
> This will need a followup in o/release to avoid breaking PRs - ideally submit a PR in advance so that the period where PRs are broken in openshift/release is as short as possible.
> 
> ```diff
>  diff --git a/core-services/prow/02_config/_plugins.yaml b/core-services/prow/02_config/_plugins.yaml
> index b7a15185..69d19175 100644
> --- a/core-services/prow/02_config/_plugins.yaml
> +++ b/core-services/prow/02_config/_plugins.yaml
> @@ -1807,8 +1807,3 @@ size:
>    xl: 0
>    xxl: 0
>  slack: {}
> -triggers:
> -- repos:
> -  - openshift/assisted-service
> -  trusted_apps:
> -  - dependabot
> diff --git a/core-services/prow/02_config/openshift/assisted-service/_pluginconfig.yaml b/core-services/prow/02_config/openshift/assisted-service/_pluginconfig.yaml
> index 2fbd7a9f..4c8a58a3 100644
> --- a/core-services/prow/02_config/openshift/assisted-service/_pluginconfig.yaml
> +++ b/core-services/prow/02_config/openshift/assisted-service/_pluginconfig.yaml
> @@ -38,3 +38,8 @@ plugins:
>      plugins:
>      - approve
>      - size
> +triggers:
> +- repos:
> +  - openshift/assisted-service
> +  trusted_apps:
> +  - dependabot 
> ```

yes, I'll revert the configurations for now to make it easier. Either way it's not critical and we want to have it in the split config files:
https://github.com/openshift/release/pull/26288

**Pull request merged**